test/integration-new/cni-version-testing/cni_version_host_networking_test.go

```diff
@@ -0,0 +1,316 @@
+)
+
+var _ = Describe("test host networking on upgrade/downgrade", func() {
+
+	It("should apply initial addon version successfully", func() {
+		ApplyAddOn(initialCNIVersion)
+		//Set the WARM_ENI_TARGET to 0 to prevent all pods being scheduled on secondary ENI
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, "aws-node", "kube-system",
+			"aws-node", map[string]string{"WARM_IP_TARGET": "3", "WARM_ENI_TARGET": "0"})
+
+	})
+
+	Context("when testing host networking on initial version", func() {
+		testHostNetworking()
+	})
+
+	It("should apply final addon version successfully", func() {
+		ApplyAddOn(finalCNIVersion)
+		//Set the WARM_ENI_TARGET to 0 to prevent all pods being scheduled on secondary ENI
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, "aws-node", "kube-system",
+			"aws-node", map[string]string{"WARM_IP_TARGET": "3", "WARM_ENI_TARGET": "0"})
+
+	})
+
+	Context("when testing host networking on final version", func() {
+		testHostNetworking()
+	})
+
+})
+
+func testHostNetworking() {
+	var err error
+	var podLabelKey = "app"
+	var podLabelVal = "host-networking-test"
+
+	Context("when pods using IP from primary and secondary ENI are created", func() {
+		AfterEach(func() {
+			k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+				AWS_VPC_ENI_MTU:            DEFAULT_MTU_VAL,
+				AWS_VPC_K8S_CNI_VETHPREFIX: DEFAULT_VETH_PREFIX,
+			})
+		})
+		It("should have correct host networking setup when running and cleaned up once terminated", func() {
+			// Launch enough pods so some pods end up using primary ENI IP and some using secondary
+			// ENI IP
+			deployment := manifest.NewBusyBoxDeploymentBuilder().
+				Replicas(maxIPPerInterface*2).
+				PodLabel(podLabelKey, podLabelVal).
+				NodeName(primaryNode.Name).
+				Build()
+
+			By("creating a deployment to launch pod using primary and secondary ENI IP")
+			deployment, err = f.K8sResourceManagers.DeploymentManager().
+				CreateAndWaitTillDeploymentIsReady(deployment, utils.DefaultDeploymentReadyTimeout)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("getting the list of pods using IP from primary and secondary ENI")
+			interfaceTypeToPodList :=
+				GetPodsOnPrimaryAndSecondaryInterface(primaryNode, podLabelKey, podLabelVal)
+
+			// Primary ENI and Secondary ENI IPs are handled differently when setting up
+			// the host networking rule hence this check
+			Expect(len(interfaceTypeToPodList.PodsOnSecondaryENI)).
+				Should(BeNumerically(">", 0))
+			Expect(len(interfaceTypeToPodList.PodsOnPrimaryENI)).
+				Should(BeNumerically(">", 0))
+
+			By("generating the pod networking validation input to be passed to tester")
+			input, err := GetPodNetworkingValidationInput(interfaceTypeToPodList).Serialize()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating host networking setup is setup correctly")
+			ValidateHostNetworking(NetworkingSetupSucceeds, input)
```

Comment:
This looks like duplicate logic. Is there a way by which we can avoid replicating this same logic in 2 places ?